src/compiler/checker.ts

```diff
@@ -8594,9 +8602,10 @@ namespace ts {
                 if (type) {
                     return type;
                 }
-                const signature = getSignatureFromDeclaration(container);
-                if (signature.thisType) {
-                    return signature.thisType;
+
+                const thisType = getThisTypeOfSignature(getSignatureFromDeclaration(container));

```

Comment:
except for the argument type, this could be a call to `getAnnotatedAccessorThisType`